tests/observability/metrics/utils.py

```diff
@@ -1342,3 +1353,86 @@ def get_vmi_guest_os_kernel_release_info_metric_from_vm(
         NODE_STR: vm.vmi.virt_launcher_pod.node.name,
         "vmi_pod": vm.vmi.virt_launcher_pod.name,
     }
+
+
+def get_pvc_size_bytes(vm: VirtualMachineForTests) -> str:
+    vm_dv_templates = vm.instance.spec.dataVolumeTemplates
+    assert vm_dv_templates or len(vm_dv_templates) > 0, "VM has no DataVolume templates"
+    return str(
+        int(
+            bitmath.parse_string_unsafe(
+                PersistentVolumeClaim(
+                    name=vm_dv_templates[0].metadata.name,
+                    namespace=vm.namespace,
+                ).instance.spec.resources.requests.storage
+            ).Byte.bytes
+        )
+    )
+
+
+def get_vm_virt_launcher_pod_requested_memory(vm: VirtualMachineForTests) -> int:
+    containers = vm.vmi.virt_launcher_pod.instance.spec.containers
+    assert containers, f"No containers found in virt-launcher pod of {vm.vmi.virt_launcher_pod.name}"
+    return int(bitmath.parse_string_unsafe(containers[0].resources.requests.memory).bytes)
+
+
+def wait_for_virt_launcher_pod_metrics_resource_exists(vm_for_test: VirtualMachineForTests) -> None:
+    vl_name = vm_for_test.vmi.virt_launcher_pod.name
+    samples = TimeoutSampler(
+        wait_timeout=TIMEOUT_1MIN,
+        sleep=TIMEOUT_15SEC,
+        func=lambda: PodMetrics(name=vl_name, namespace=vm_for_test.namespace, client=get_client()).exists,
```

Comment:
`client=get_client()` - either use `vm.client` or pass client to the function.
if a resource does not get `client`, it will use `get_client()` and this is risky of being implicit and not necessarily use the client we want.